src/App.vue

```diff
@@ -1,23 +1,103 @@
+
+
 <template>
   <div id="app">
-    <img src="./assets/logo.png">
-    <router-view/>
+    <toolbar></toolbar>
+    <router-view :meals="meals" @mealUsed="markUsed"/>
   </div>
 </template>
 
 <script>
+  import { map, pipe, curry, trace } from "@/utils/fp";
+  import MealService from "@/services/meals";
+  import mediator from "@/mediator";
+
+  import Toolbar from "@/components/Toolbar.vue"
+
+  const identifyMeal = (meal, id) => Object.assign({id}, meal);
+
+  const unIdentifyMeal = function(meal) {
+  	let unidentifiedMeal = {};
+
+    for(let prop in meal) {
+    	if(meal.hasOwnProperty(prop) && prop !== "id") {
+    		unidentifiedMeal[prop] = meal[prop];
+      }
+    }
+    return unidentifiedMeal
+  }
+
+  const updateMeal = curry((changes, meal) => Object.assign(meal, changes));
+
+  const changeAtIndex = curry(function(fx, targetIdx, target, currentIdx) {
+  	if(target.id === targetIdx) {
+  		return fx(target);
+    } else {
+  		return target
+    }
+  });
```

Comment:
Proof that you can use one of these frameworks and still write really lousy code :)